File: Lesson9/app/src/main/java/ru/mirea/khokhlov/Lesson9/MainActivity.java
package ru.mirea.khokhlov.Lesson9;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import ru.mirea.khokhlov.data.Storage.MovieStorage;
import ru.mirea.khokhlov.data.Storage.SharedPrefMovieStorage;
import ru.mirea.khokhlov.data.repository.MovieRepositoryImpl;
import ru.mirea.khokhlov.domain.domain.Models.Movie;
import ru.mirea.khokhlov.domain.domain.repository.MovieRepository;
import ru.mirea.khokhlov.domain.domain.usecases.GetFavoriteFilmUseCase;
import ru.mirea.khokhlov.domain.domain.usecases.SaveMovieToFavoriteUseCase;
import com.example.rumireaprivalovlesson9.R;

public class MainActivity extends AppCompatActivity {

    private MainViewModel vm;
    private MovieRepository movieRepository;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Инициализация MovieStorage
        MovieStorage movieStorage = new SharedPrefMovieStorage(this);

        // Инициализация репозитория
        movieRepository = new MovieRepositoryImpl(movieStorage);

        // Получаем элементы интерфейса
        EditText editTextMovie = findViewById(R.id.editTextMovie);
        TextView textViewMovie = findViewById(R.id.textViewMovie);

        // Устанавливаем слушатель для кнопки "Сохранить фильм"
        findViewById(R.id.buttonSaveMovie).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String movieName = editTextMovie.getText().toString();
                Boolean result = new SaveMovieToFavoriteUseCase(movieRepository).execute(new Movie(2, movieName));
                textViewMovie.setText(String.format("Save result: %s", result));
            }
        });

        // Устанавливаем слушатель для кнопки "Получить фильм"
        findViewById(R.id.buttonGetMovie).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Movie movie = new GetFavoriteFilmUseCase(movieRepository).execute();
                textViewMovie.setText(String.format("Movie name: %s", movie.getName()));
            }
        });

        Log.d(MainActivity.class.getSimpleName(), "MainActivity created");
        vm = new ViewModelProvider(this, new ViewModelFactory(this)).get(MainViewModel.class);
    }
}

File: Lesson3/sharer/src/main/java/ru/mirea/hohlovdv/sharer/MainActivity.kt
package ru.mirea.hohlovdv.sharer

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        var btnSharer = findViewById<Button>(R.id.btnSharer)
        btnSharer.setOnClickListener{
            startActivity (Intent(this, Sharer::class.java))
        }
        var btnPick = findViewById<Button>(R.id.btnPick)
        btnPick.setOnClickListener{
            startActivityForResult(Intent(this, Pick::class.java) , 200)
        }

    }
}